File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.go
func findSpecialInteger(arr []int) int {
    var quarter int
    quarter = len(arr)/4
    cnt := 1
    
    for i:=1; i<len(arr); i++ {
        if arr[i-1] == arr[i] {
            cnt++
            if cnt>quarter {
                return arr[i-1]
            }
        } else {
            cnt = 1
        }
    }
    return arr[0]
}

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        int quarter = arr.size()/4;
        int cnt=1;
        
        for(int i=1; i<arr.size(); i++) {
            if(arr[i-1]==arr[i]) {
                cnt++;
                if(cnt>quarter) 
                    return arr[i-1];
            }
            else {
                cnt = 1;
            }
        }
        return arr[0];
    }
};